File: app/src/main/java/com/battlelancer/seriesguide/ui/people/PersonActivity.java
package com.battlelancer.seriesguide.ui.people;

import android.os.Bundle;
import android.view.MenuItem;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.core.content.ContextCompat;
import com.battlelancer.seriesguide.R;
import com.battlelancer.seriesguide.ui.BaseActivity;

/**
 * Hosts a {@link PersonFragment}, only used on handset devices. On
 * tablet-size devices a two-pane layout inside {@link PeopleActivity}
 * is used.
 */
public class PersonActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        // instead of Window.FEATURE_ACTION_BAR_OVERLAY as indicated by AppCompatDelegate warning
        supportRequestWindowFeature(AppCompatDelegate.FEATURE_SUPPORT_ACTION_BAR_OVERLAY);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_person);
        setupActionBar();

        if (savedInstanceState == null) {
            PersonFragment f = PersonFragment.newInstance(
                    getIntent().getIntExtra(PersonFragment.ARG_PERSON_TMDB_ID, 0));
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.containerPerson, f)
                    .commit();
        }
    }

    @Override
    protected void setupActionBar() {
        super.setupActionBar();
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setBackgroundDrawable(
                    ContextCompat.getDrawable(this, R.drawable.background_actionbar_gradient));
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowTitleEnabled(false);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int itemId = item.getItemId();
        if (itemId == android.R.id.home) {
            onBackPressed();
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/battlelancer/seriesguide/ui/people/PersonActivity.kt
package com.battlelancer.seriesguide.ui.people

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatDelegate
import androidx.core.content.ContextCompat
import com.battlelancer.seriesguide.R
import com.battlelancer.seriesguide.ui.BaseActivity

/**
 * Hosts a [PersonFragment], only used on handset devices. On
 * tablet-size devices a two-pane layout inside [PeopleActivity]
 * is used.
 */
class PersonActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        // Instead of Window.FEATURE_ACTION_BAR_OVERLAY as indicated by AppCompatDelegate warning.
        supportRequestWindowFeature(AppCompatDelegate.FEATURE_SUPPORT_ACTION_BAR_OVERLAY)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_person)
        setupActionBar()

        if (savedInstanceState == null) {
            val f = PersonFragment.newInstance(
                intent.getIntExtra(PersonFragment.ARG_PERSON_TMDB_ID, 0)
            )
            supportFragmentManager.beginTransaction()
                .add(R.id.containerPerson, f)
                .commit()
        }
    }

    override fun setupActionBar() {
        super.setupActionBar()
        supportActionBar?.also {
            it.setBackgroundDrawable(
                ContextCompat.getDrawable(this, R.drawable.background_actionbar_gradient)
            )
            it.setHomeAsUpIndicator(R.drawable.ic_clear_24dp)
            it.setDisplayHomeAsUpEnabled(true)
            it.setDisplayShowTitleEnabled(false)
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                onBackPressed()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}
